### PR TITLE
Add chips to search filter

### PR DESCRIPTION
**Pull request by jmarrero**

### CODE REVIEW

File: frontend/src/app/search.tsx [hunk 1]
```diff
@@ -79,6 +81,7 @@ class Search extends Component<IAppState, ISearchState> {
 
     return (
       <React.Fragment>
+        {console.log("This is the query: " + this.state.filterQuery)}
```
Comment (jmarrero) [inline frontend/src/app/search.tsx]: This is here for testing purposes, this PR is not ready.

**Comment by jmarrero** [comment]
Test are missing but otherwise this is ready for review @xdavidson

File: pantheon-bundle/frontend/src/app/search.tsx [hunk 2]
```diff
@@ -122,10 +118,10 @@ class Search extends Component<IAppState, ISearchState> {
                   <DataListItemCells
                     dataListCells={[
                       <DataListCell width={2} key="title">
-                        <button onClick={this.sortByName} className="sp-prop" id="span-name" aria-label="sort column by name">Name</button>
+                        <span className="sp-prop-nosort" id="span-name" aria-label="column name">Name</span>
                       </DataListCell>,
                       <DataListCell width={2} key="description">
-                        <button onClick={this.sortByDescription} className="sp-prop" id="span-name" aria-label="sort column by description">Description</button>
+                        <span className="sp-prop-nosort" id="span-name" aria-label="column description">Description</span>
```
Comment (xdavidson) [inline pantheon-bundle/frontend/src/app/search.tsx]: The span tag id looks like a typo. It's a duplicate of the Name cell above. Should it be id="span-description" ?

File: pantheon-bundle/frontend/src/app/search.tsx [hunk 3]
```diff
@@ -134,7 +130,7 @@ class Search extends Component<IAppState, ISearchState> {
                         <span className="sp-prop-nosort" id="span-source-name">Source Name</span>
                       </DataListCell>,
                       <DataListCell key="upload time">
-                        <button onClick={this.sortByUploadTime} className="sp-prop" id="span-name" aria-label="sort column by upload time">Upload Time</button>
+                        <span className="sp-prop-nosort" id="span-name" aria-label="column upload time">Upload Time</span>
```
Comment (xdavidson) [inline pantheon-bundle/frontend/src/app/search.tsx]: duplicated span id here as well.

File: pantheon-bundle/frontend/src/app/searchFilter.tsx [hunk 4]
```diff
@@ -1,36 +1,381 @@
...
+
+    this.setState({ chipGroups: copyOfChipGroups, productsQueryParam: prodQuery, productversionsQueryParam: verQuery }, () => {
+      this.setQuery();
+    });
+  };
+
+  // Should be called after each change of state
+  private setQuery = () => {
+    let searchQuery = ""
+    if (this.state.searchText !== "") {
```
Comment (xdavidson) [inline pantheon-bundle/frontend/src/app/searchFilter.tsx]: I suggest we use (mystring.trim() !== "") to evaluate if mystring is empty or not. trim removes whitespace and newlines

File: pantheon-bundle/frontend/src/app/versions.tsx [hunk 5]
```diff
@@ -663,7 +663,7 @@ class Versions extends Component<IProps, any> {
                             this.setState({ moduleUrl: url })
                         }
                         this.setState({
-                            usecaseValue: this.state.metadataResults.documentUsecase,
+                            usecaseValue: this.state.metadataResults.documentUseCase,
```
Comment (xdavidson) [inline pantheon-bundle/frontend/src/app/versions.tsx]: In metadataResults, the fieldname is 'documentUsercase', by changing it to upper case 'C', edit metadata no longer works.

File: pantheon-bundle/frontend/src/app/searchFilter.tsx [hunk 6]
```diff
@@ -1,36 +1,381 @@
...
+      searchQuery += "search=" + this.state.searchText
+    }
+
+    if (this.state.productsQueryParam !== "") {
+      if (searchQuery !== "") {
```
Comment (xdavidson) [inline pantheon-bundle/frontend/src/app/searchFilter.tsx]: I suggest we add .trim() to the evaluation. Currently when I search a string with extra space in the end, it returns empty results. If I remove the trailing space in my search string, i see matching results.

File: pantheon-bundle/frontend/src/app/versions.tsx (hunk 5 above)
Comment (jmarrero) [reply pantheon-bundle/frontend/src/app/versions.tsx]: @xdavidson I think in line 498  is the place where we append the metadata, I don't see anywhere in the backend where we are defining it other than doing the Query. I think I changed all the occurrences on the code from Usecase to UseCase. Not sure if I am missing somewhere else I need to change it.

File: pantheon-bundle/frontend/src/app/searchFilter.tsx (hunk 6 above)
Comment (jmarrero) [reply pantheon-bundle/frontend/src/app/searchFilter.tsx]: great points! I am adding trim() to them.

File: pantheon-bundle/frontend/src/app/search.tsx (hunk 2 above)
Comment (jmarrero) [reply pantheon-bundle/frontend/src/app/search.tsx]: yep overlooked those ids, they have been updated in the latest commit too.

**Comment by carlosmunoz** [comment]
@jmarrero just wanted to point out this PR was merged while lowering our code coverage.

**Pull request merged**